Mathlib/Data/List/Basic.lean

```diff
@@ -1231,6 +1231,12 @@ theorem forall_cons (p : α → Prop) (x : α) : ∀ l : List α, Forall p (x ::
   | [] => (and_iff_left_of_imp fun _ ↦ trivial).symm
   | _ :: _ => Iff.rfl
 
+@[simp]
+theorem forall_append {p : α → Prop} : ∀ {xs ys : List α},
```

Comment:
We do have [`List.forall_mem_append`](https://leanprover-community.github.io/mathlib4_docs/Init/Data/List/Lemmas.html#List.forall_mem_append), but that is stated sufficiently differently that I think this proof is the best way to go.